### ISSUE TITLE
Double templating again

### ISSUE THREAD

**Issue by willthames** [ansible/ansible#65898]
##### SUMMARY
Yet another double templating regression

The circumstances are different but the test case is exactly the same as  #62237. The fix should include adding this as a regression test case. 

cc @sivel as the breaking commit is 254788b and the circumstances don't seem to be resolved by the documented porting guide.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
templating

##### ANSIBLE VERSION

```paste below
ansible 2.9.2
  config file = None
  configured module search path = ['/Users/wthames/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.7.5 (default, Nov  1 2019, 02:16:23) [Clang 11.0.0 (clang-1100.0.33.8)]
```

##### CONFIGURATION

```paste below
```
None

##### OS / ENVIRONMENT

N/A

##### STEPS TO REPRODUCE
Run ansible-playbook play.yml -vv with the contents of the following gist:

https://gist.github.com/willthames/994b3f3d7a5b43ecf3a0855d82857f3d

##### EXPECTED RESULTS

```
ansible-playbook 2.9.0.post0
  config file = None
  configured module search path = ['/Users/wthames/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/wthames/src/opensource/ansible/lib/ansible
  executable location = /Users/wthames/src/opensource/ansible/bin/ansible-playbook
  python version = 3.7.5 (default, Nov  1 2019, 02:16:23) [Clang 11.0.0 (clang-1100.0.33.8)]
No config file found; using defaults
[WARNING]: No inventory was parsed, only implicit localhost is available

[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

/usr/local/lib/python3.7/site-packages/requests/__init__.py:91: RequestsDependencyWarning: urllib3 (1.25.7) or chardet (3.0.4) doesn't match a supported version!
  RequestsDependencyWarning)

PLAYBOOK: play.yml ***********************************************************************************************************************************************************************************************************************************
1 plays in play.yml

PLAY [localhost] *************************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] *******************************************************************************************************************************************************************************************************************************
task path: /Users/wthames/tmp/ansible/unsafe_proxy/play.yml:1
ok: [localhost]
META: ran handlers

TASK [set_fact] **************************************************************************************************************************************************************************************************************************************
task path: /Users/wthames/tmp/ansible/unsafe_proxy/play.yml:4
ok: [localhost] => {"ansible_facts": {"resources": []}, "changed": false}

TASK [set_fact] **************************************************************************************************************************************************************************************************************************************
task path: /Users/wthames/tmp/ansible/unsafe_proxy/play.yml:7
ok: [localhost] => (item=includer.yml) => {"ansible_facts": {"resources": [{"apiVersion": "v1", "data": {"node.rules.yml": "groups:\n  - name: node.rules\n    rules:\n    - alert: node.loadaverage\n      expr: (node_load5 / count(node_cpu_seconds_total{job=\"aws-nodes\",mode=\"system\"}) WITHOUT (cpu, mode)) > 20\n      for: 2m\n      labels:\n        service: k8s\n        severity: INFO\n      annotations:\n        description: '{{ $labels.instance }}: The 5 minute average load is high: {{ $value | printf \"%.2f\" }}'\n        summary: '{{ $labels.instance }}: High load average detected'"}, "kind": "ConfigMap", "metadata": {"name": "test-rules", "namespace": "monitoring"}}]}, "ansible_loop_var": "item", "changed": false, "item": "includer.yml"}

TASK [include_tasks] *********************************************************************************************************************************************************************************************************************************
task path: /Users/wthames/tmp/ansible/unsafe_proxy/play.yml:13
included: /Users/wthames/tmp/ansible/unsafe_proxy/task.yml for localhost

TASK [debug] *****************************************************************************************************************************************************************************************************************************************
task path: /Users/wthames/tmp/ansible/unsafe_proxy/task.yml:1
ok: [localhost] => {
    "item": {
        "apiVersion": "v1",
        "data": {
            "node.rules.yml": "groups:\n  - name: node.rules\n    rules:\n    - alert: node.loadaverage\n      expr: (node_load5 / count(node_cpu_seconds_total{job=\"aws-nodes\",mode=\"system\"}) WITHOUT (cpu, mode)) > 20\n      for: 2m\n      labels:\n        service: k8s\n        severity: INFO\n      annotations:\n        description: '{{ $labels.instance }}: The 5 minute average load is high: {{ $value | printf \"%.2f\" }}'\n        summary: '{{ $labels.instance }}: High load average detected'"
        },
        "kind": "ConfigMap",
        "metadata": {
            "name": "test-rules",
            "namespace": "monitoring"
        }
    }
}
META: ran handlers
META: ran handlers

PLAY RECAP *******************************************************************************************************************************************************************************************************************************************
localhost                  : ok=5    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

##### ACTUAL RESULTS

```paste below
ansible-playbook 2.9.0.post0
  config file = None
  configured module search path = ['/Users/wthames/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/wthames/src/opensource/ansible/lib/ansible
  executable location = /Users/wthames/src/opensource/ansible/bin/ansible-playbook
  python version = 3.7.5 (default, Nov  1 2019, 02:16:23) [Clang 11.0.0 (clang-1100.0.33.8)]
No config file found; using defaults
[WARNING]: No inventory was parsed, only implicit localhost is available

[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

/usr/local/lib/python3.7/site-packages/requests/__init__.py:91: RequestsDependencyWarning: urllib3 (1.25.7) or chardet (3.0.4) doesn't match a supported version!
  RequestsDependencyWarning)

PLAYBOOK: play.yml ***********************************************************************************************************************************************************************************************************************************
1 plays in play.yml

PLAY [localhost] *************************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] *******************************************************************************************************************************************************************************************************************************
task path: /Users/wthames/tmp/ansible/unsafe_proxy/play.yml:1
ok: [localhost]
META: ran handlers

TASK [set_fact] **************************************************************************************************************************************************************************************************************************************
task path: /Users/wthames/tmp/ansible/unsafe_proxy/play.yml:4
ok: [localhost] => {"ansible_facts": {"resources": []}, "changed": false}

TASK [set_fact] **************************************************************************************************************************************************************************************************************************************
task path: /Users/wthames/tmp/ansible/unsafe_proxy/play.yml:7
ok: [localhost] => (item=includer.yml) => {"ansible_facts": {"resources": [{"apiVersion": "v1", "data": {"node.rules.yml": "groups:\n  - name: node.rules\n    rules:\n    - alert: node.loadaverage\n      expr: (node_load5 / count(node_cpu_seconds_total{job=\"aws-nodes\",mode=\"system\"}) WITHOUT (cpu, mode)) > 20\n      for: 2m\n      labels:\n        service: k8s\n        severity: INFO\n      annotations:\n        description: '{{ $labels.instance }}: The 5 minute average load is high: {{ $value | printf \"%.2f\" }}'\n        summary: '{{ $labels.instance }}: High load average detected'"}, "kind": "ConfigMap", "metadata": {"name": "test-rules", "namespace": "monitoring"}}]}, "ansible_loop_var": "item", "changed": false, "item": "includer.yml"}

TASK [include_tasks] *********************************************************************************************************************************************************************************************************************************
task path: /Users/wthames/tmp/ansible/unsafe_proxy/play.yml:13
fatal: [localhost]: FAILED! => {"msg": "An unhandled exception occurred while templating '[{'apiVersion': 'v1', 'kind': 'ConfigMap', 'metadata': {'name': 'test-rules', 'namespace': 'monitoring'}, 'data': {'node.rules.yml': 'groups:\\n  - name: node.rules\\n    rules:\\n    - alert: node.loadaverage\\n      expr: (node_load5 / count(node_cpu_seconds_total{job=\"aws-nodes\",mode=\"system\"}) WITHOUT (cpu, mode)) > 20\\n      for: 2m\\n      labels:\\n        service: k8s\\n        severity: INFO\\n      annotations:\\n        description: \\'{{ $labels.instance }}: The 5 minute average load is high: {{ $value | printf \"%.2f\" }}\\'\\n        summary: \\'{{ $labels.instance }}: High load average detected\\''}}]'. Error was a <class 'ansible.errors.AnsibleError'>, original message: template error while templating string: unexpected char '$' at 299. String: groups:\n  - name: node.rules\n    rules:\n    - alert: node.loadaverage\n      expr: (node_load5 / count(node_cpu_seconds_total{job=\"aws-nodes\",mode=\"system\"}) WITHOUT (cpu, mode)) > 20\n      for: 2m\n      labels:\n        service: k8s\n        severity: INFO\n      annotations:\n        description: '{{ $labels.instance }}: The 5 minute average load is high: {{ $value | printf \"%.2f\" }}'\n        summary: '{{ $labels.instance }}: High load average detected'"}

PLAY RECAP *******************************************************************************************************************************************************************************************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```

**Comment by willthames** [comment]
This may not be a bug although the porting guide could possibly be clearer. 

It's clear from the porting guide that changing from with_X to loop might fix problems, but in my case, changing from loop to with_X fixed the problem.

**Comment by willthames** [comment]
Also, breaking changes in 2.9 aren't really cool.

This worked fine in 2.9.0 but upgrading to 2.9.2 broke it for us.

**Comment by gundalow** [comment]
Need to add very clear tests for this and ensure they are backported to (at least) stable-2.9

**Comment by sivel** [comment]
The current behavior of the 2.9.1 release (and newer) is actually correct.

In 2.9 we cleaned up some invalid assumptions about how unsafe was handled.  Unfortunately, we missed a place, and `loop` was incorrectly being treated as unsafe.  This was subsequently resolved in the 2.9.1 release.  `loop` doesn't do any data manipulation, so we rely on the data passed to `loop` to be marked as unsafe.

Since `with_X` is powered by lookups, and lookups are considered unsafe, we ensure that data produced by `with_X` is properly marked as unsafe.

In your case, you are using `lookup`, which is returning unsafe data, but we don't have a way to say that manipulations to unsafe data should remain unsafe.

As such, your templating action of the following causes the unsafe marker to be removed from the data making it "safe", and thus subject to templating:

```
resources + lookup('template', item) | from_yaml_all | list
```

In this case, since we don't have an `unsafe` filter, which I think might be a good addition, the only way to resolve your issue, is in the manner in which you have resolved it, by switching to `with_items`.

As such, there is really nothing to do here, maybe other than detailing an additional scenario in the porting guide.

Tests were added to ensure that functionality is working as expected, so there really is no need for additional tests.
